zero_bin/rpc/src/main.rs

```diff
@@ -56,6 +56,13 @@ impl Cli {
             } => {
                 let checkpoint_block_number =
                     checkpoint_block_number.unwrap_or((start_block - 1).into());
+                if checkpoint_block_number != (start_block - 1).into() {
```

Comment:
This is invalid check. `checkpoint_block_unmber` could be smaller than start block. If it is not provided, we use it as default `start_block-1`